docs/collection-tables.rst

```diff
@@ -0,0 +1,89 @@
+.. _collection-tables:
+
+=================
+Collection Tables
+=================
+
+Imagine you are modelling gene expression data from a given collection.
+Ideally, you would want all expression values organized in a table where
+rows represents samples and columns represent genes. Class
+``CollectionTables`` gives you just that (and more).
+
+We will present the functionality of ``CollectionTables`` through an
+example. We will:
+
+- Create an instance of ``CollectionTables`` and examine it's attributes
+- Fetch raw expression counts and only keep `TIS genes`_ with sufficient
+  expression
+- Compute ``log2(TPM + 1)`` of expression values and visualize samples
+  in a simple PCA plot
+
+.. _`TIS genes`: https://translational-medicine.biomedcentral.com/articles/10.1186/s12967-019-2100-3
+
+First, connect to a Resolwe server, pick a collection and create
+and instance of ``CollectionTables``::
+
+    import resdk
+    res = resdk.Resolwe(url='https://app.genialis.com/')
+    res.login()
+    collection = res.collection.get("sum149-fresh-for-rename")
+    ct = resdk.CollectionTables(collection)
+
+Object ``ct`` is an instance of ``CollectionTables`` and has many attributes. For a complete list see
+the :ref:`reference`, here we list the most commonly used ones::
+
+    # Expressions raw counts
+    ct.rc
+
+    # Expressions normalized counts
+    ct.exp
+    # See normalization method
+    ct.exp.attrs["exp_type"]
+
+    # Sample metadata
+    ct.meta
+
+    # Dictionary that maps gene ID's into gene symbols
+    ct.id_to_symbol
+    # This is handy to rename column names (gene ID's) to gene symbols
+    ct.rc.rename(columns=ct.id_to_symbol)
+
+
+.. note::
+
+  Expressions and metadata are cached in memory as well as on disk. At
+  each time they are re-requested a check is made that local and server side
+  of data is synced. If so, cached data is used. Otherwise, new data
+  will be pulled from server.
+
+In our example we will only work with a set of `TIS genes`_::
+
+    TIS_GENES = ["CD3D", "IDO1", "CIITA", "CD3E", "CCL5", "GZMK", "CD2", "HLA-DRA", "CXCL13", "IL2RG", "NKG7", "HLA-E", "CXCR6", "LAG3", "TAGAP", "CXCL10", "STAT1", "GZMB"]
+
+We will identify low expressed genes and only keep the ones with average raw
+expression above 1::
```

Comment:
Above 20 make more sense, but I assume you don't get enough genes with sufficient coverage then.

Reply:
Better bad PCA plot than introducing bad filtering practice. Fixed to 20.

Reply:
How many genes are you left with?

Reply:
Five (5)

Reply:
If i set threshold to 1 then 11 genes remain.

Reply:
I assume that's because we have a low number of samples... Five is okay.